core/node/api_server/src/tx_sender/mod.rs

```diff
@@ -368,7 +390,7 @@ impl TxSender {
         let submission_res_handle = self
             .0
             .tx_sink
-            .submit_tx(&tx, execution_output.metrics)
+            .submit_tx(&tx, execution_output.metrics, validation_result.unwrap())
```

Comment:
Please replace the `let Err(_) = validation_result` to make it clear that this is safe:

```rust
let validation_traces = validation_result?;
// ...
.submit_tx(&tx, execution_output.metrics, validation_traces)
```